appstats.js

```diff
@@ -232,60 +232,78 @@ class AzureAppInsightStats extends AzureAppStats {
     }
 
     getFunctionStats(functionName, timestamp, callback) {
-        if (process.env.APPINSIGHTS_INSTRUMENTATIONKEY || process.env.APPLICATIONINSIGHTS_CONNECTION_STRING) {
-            const managementClient = new ApplicationInsightsManagementClient(new DefaultAzureCredential(), this.subscriptionId);
-            managementClient.components.listByResourceGroup(this.resourceGroup).then((result) => {
-                const query = {
-                    "query": `requests
-                            | where operation_Name =~ '${functionName}'
-                            | order by timestamp desc
-                            | where success == "True" or success == "False"
-                            | summarize errors = countif(success == "False"),invocations = countif(success == "True" or success == "False") by operation_Name
-                            | extend details = pack_all()
-                            | summarize Result = make_list(details)`, timespan: 'PT15M'
-                };
-                const insightsClient = new ApplicationInsightsDataClient(this.tokenCredentials, { subscriptionId: this.subscriptionId });
-
-                insightsClient.query.execute(result[0].appId, query).then((result) => {
-                    try {
-                        const data = JSON.parse(result.tables[0].rows[0]);
-                        if (data.length) {
-                            const dataObj = { [data[0].operation_Name]: { invocations: data[0].invocations, errors: data[0].errors } };
-                            return callback(null, dataObj);
-                        } else {
-                            const dataObj = { [functionName]: { invocations: 0, errors: 0 } };
-                            return callback(null, dataObj);
-                        }
-                    } catch (err) {
-                        let errMessage = `An error occurred, while getting statistics ${err}`;
-                        return callback(errMessage, null);
+        super.getFunctionStats(functionName, timestamp, callback);
+    }
+    
+    getAppInsightsFunctionStats(functionNames, timestamp, callback) {
+        var appstats = this;
+        const managementClient = new ApplicationInsightsManagementClient(new DefaultAzureCredential(), this.subscriptionId);
+        managementClient.components.listByResourceGroup(this.resourceGroup).then((result) => {
+            const insightsClient = new ApplicationInsightsDataClient(this.tokenCredentials, { subscriptionId: this.subscriptionId });
+            let query = {
+                "query": `requests
+                    | where operation_Name =~ "${functionNames[0]}" or  operation_Name =~ "${functionNames[1]}" or 
+                     operation_Name =~ "${functionNames[2]}" or operation_Name =~ "${functionNames[3]}" 
```

Comment:
can we please generate this condition based on how many names are in `functionNames`? Or change this condition somehow (like, `operation_Name in [${functionNames}] ` ) to avoid changing this code every time we change the number of functions used by a collector

Reply:
as per suggestions modified condition to use **in** operator